[revert the minor PR45144 missed-optimization fix because it
 results in miscompilation of gnat on m68k with gcc-4.6 and 4.5;
 with gcc-4.7 other changes mask the issue <mikpe@it.uu.se>]

	PR ada/48835

--- a/src/gcc/tree-sra.c
+++ b/src/gcc/tree-sra.c
@@ -916,9 +916,6 @@ type_consists_of_records_p (tree type)
       {
 	tree ft = TREE_TYPE (fld);
 
-	if (DECL_BIT_FIELD (fld))
-	  return false;
-
 	if (!is_gimple_reg_type (ft)
 	    && !type_consists_of_records_p (ft))
 	  return false;
